core/src/program/tests.rs

```diff
@@ -0,0 +1,42 @@
+use super::{Deserializable, Digest, Felt, Kernel, ProgramInfo, Serializable};
+use crate::Word;
+use proptest::prelude::*;
+use rand_utils::prng_array;
+
+proptest! {
+    #[test]
+    fn arbitrary_program_info_serialization_works(
+        kernel_count in prop::num::u8::ANY,
+        ref seed in any::<[u8; 32]>()
```

Comment:
wouldn't this work with something like `any::<u64; 4>`? we should probably make this into a separate strategy on `crypto` later on.

Reply:
This would work for a single digest. What we do there is build a stream of digests from a given seed by randomizing it at every iteration

Reply:
The digests count could also be `0`, that is a very important edge case for serialization. `proptest` will definitely send a `kernel_count` `0` as it often breaks most of implementations

Reply:
humm, is that a good idea? it would throw the simplify step of the tool off, no?

Reply:
I'm not sure wdym. One alternative is to implement [Arbitrary](https://altsysrq.github.io/rustdoc/proptest/1.0.0/proptest/prelude/trait.Arbitrary.html) for each of our types and take a vector of these.

However, we would have to think of a way of not getting very big amounts of digests. proptest would just create random size vectors, and any `usize` count of digests might be just too much for a normal desktop to handle. We do this "manual u8 digests count" because the worst case scenario, that is 255 digests, is trivial

Reply:
They do claim however that they take the size of the type into consideration when computing the length of the vector so they will not create outrageously big data sets, but I remember having bad experiences with this in the past.

Reply:
Just to clarfiy, for this case, it probably doesn't matter, it is a simple serde test. I'm fine with merging as it is. For some more complicated proptest, then I would definitely prefer to have better shrinking without tying up the generation of different objects.

Reply:
This is indeed a simple case. However, in general, we might indeed benefit from an engine for arbitrary sets to be used as arguments for fuzzy-tests. At least for our most atomic types (word, digest, etc).

Our general needs are simple: stream of random, uniformly distributed words. Anything we have can be derived from a word, so that should suffice.